instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java

```diff
@@ -289,6 +305,40 @@ private void onSdkResponse(
     }
   }
 
+  @Override
+  public void afterTransmission(
+      Context.AfterTransmission context, ExecutionAttributes executionAttributes) {
+    if (!recordIndividualHttpError) {
```

Comment:
Can we extract all this logic into a method whose name deliver the intent?

maybe

```
extractErrorMessageAsEvent
```

Reply:
Sure, I will make the change in next commit.